avro4s-core/src/main/scala/com/sksamuel/avro4s/ImmutableRecord.scala

```diff
@@ -10,7 +10,7 @@ import org.apache.avro.specific.SpecificRecord
   */
 trait Record extends GenericRecord with SpecificRecord
 
-case class ImmutableRecord(schema: Schema, values: Vector[AnyRef]) extends Record {
```

Comment:
Widening the type allows using an array on record creation which is quite efficient